### PR TITLE
stages/authenticator: use RBAC for devices API

### PR DESCRIPTION
**Pull request by rissson**
## Details

REPLACE ME

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

File: authentik/core/api/devices.py [hunk 1]
```diff
@@ -52,7 +53,7 @@ class AdminDeviceViewSet(ViewSet):
     """Viewset for authenticator devices"""
 
     serializer_class = DeviceSerializer
-    permission_classes = [IsAdminUser]
+    permission_classes = [IsAuthenticated]
```
Comment (BeryJu) [inline authentik/core/api/devices.py:56]: this should probably also use the default `ObjectPermissions`, no?

Comment (rissson) [reply authentik/core/api/devices.py:56]: I'd love to do that, but there's no way to get a queryset to ObjectPermissions because Device is abstract

Comment (rissson) [reply authentik/core/api/devices.py:56]: Also that permission class isn't actually used because we override it in the list method

**Review by BeryJu — COMMENTED** [review]
Please add tests otherwise LGTM

File: authentik/core/tests/test_devices_api.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+from json import loads
+
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from authentik.core.tests.utils import create_test_admin_user, create_test_user
+
+
+class TestApplicationsAPI(APITestCase):
```
Comment (BeryJu) [inline authentik/core/tests/test_devices_api.py:11]:
```suggestion
class TestDevicesAPI(APITestCase):
```

Comment (rissson) [reply authentik/core/tests/test_devices_api.py:11]: I don't know what you mean, i absolutely didn't copy other tests and modify them 😇

**Review by BeryJu — APPROVED** [review]

**Pull request merged**